cpp/tests/binaryop/binop-integration-test.cpp

```diff
@@ -1841,6 +1841,148 @@ TEST_F(BinaryOperationIntegrationTest, ShiftRightUnsigned_Scalar_Vector_SI64_SI6
+                                                  data_type(experimental::type_to_id<TypeOut>()));
+
+  auto out_h    = cudf::test::to_host<TypeOut>(*out);
+  auto out_data = out_h.first;
+  EXPECT_TRUE(out_data[0] > 0);
+  EXPECT_TRUE(out_data[1] < 0);
+  EXPECT_TRUE(out_data[2] > 0);
+  ASSERT_BINOP<TypeOut, TypeLhs, TypeRhs>(*out, lhs, rhs, PMOD());
+}
+
+TEST_F(BinaryOperationIntegrationTest, PMod_Vector_Scalar_FP64)
+{
+  using TypeOut = double;
+  using TypeLhs = double;
+  using TypeRhs = double;
+
+  using PMOD = cudf::library::operation::PMod<TypeOut, TypeLhs, TypeRhs>;
+
+  auto lhs = fixed_width_column_wrapper<TypeLhs>{{90770.74881, -15456.4335, 32213.22119}};
+  auto rhs = cudf::experimental::scalar_type_t<TypeRhs>(-86099.68377);
+
+  auto out = cudf::experimental::binary_operation(lhs,
+                                                  rhs,
+                                                  cudf::experimental::binary_operator::PMOD,
+                                                  data_type(experimental::type_to_id<TypeOut>()));
+
+  auto out_h    = cudf::test::to_host<TypeOut>(*out);
+  auto out_data = out_h.first;
+  EXPECT_TRUE(out_data[0] > 0);
+  EXPECT_TRUE(out_data[1] < 0);
+  EXPECT_TRUE(out_data[2] > 0);
+  ASSERT_BINOP<TypeOut, TypeLhs, TypeRhs>(*out, lhs, rhs, PMOD());
+}
+
+TEST_F(BinaryOperationIntegrationTest, PMod_Vector_Vector_FP64_FP32_FP64)
+{
+  using TypeOut = double;
+  using TypeLhs = float;
+  using TypeRhs = double;
+
+  using PMOD = cudf::library::operation::PMod<TypeOut, TypeLhs, TypeRhs>;
+
+  auto lhs = fixed_width_column_wrapper<TypeLhs>{
+    {24854.55893, 79946.87288, -86099.68377, -86099.68377, 1.0, 1.0, -1.0, -1.0}};
+  auto rhs = fixed_width_column_wrapper<TypeRhs>{{90770.74881,
+                                                  -15456.4335,
+                                                  36223.96138,
+                                                  -15456.4335,
+                                                  2.1336193413893147E307,
+                                                  -2.1336193413893147E307,
+                                                  2.1336193413893147E307,
+                                                  -2.1336193413893147E307}};
+
+  auto out = cudf::experimental::binary_operation(lhs,
+                                                  rhs,
+                                                  cudf::experimental::binary_operator::PMOD,
+                                                  data_type(experimental::type_to_id<TypeOut>()));
+
+  auto out_h    = cudf::test::to_host<TypeOut>(*out);
+  auto out_data = out_h.first;
+  EXPECT_TRUE(out_data[0] > 0);
+  EXPECT_TRUE(out_data[1] > 0);
+  EXPECT_TRUE(out_data[2] > 0);
+  EXPECT_TRUE(out_data[3] < 0);
+  EXPECT_TRUE(out_data[4] == 1);
+  EXPECT_TRUE(out_data[5] == 1);
+  EXPECT_TRUE(out_data[6] == 0);
+  EXPECT_TRUE(out_data[7] == 0);
```

Comment:
If you want to manually define values rather than use randomly defined ones, there's better mechanisms to do it. You can define another column of expected values and use column equality like this:
https://github.com/rapidsai/cudf/blob/71d645c0d2782981697bd15333717e99ba37ee70/cpp/tests/search/search_test.cpp#L108-L119

Reply:
done